### PR TITLE
Abbyy Live server deployment

### PR DESCRIPTION
**Pull request by willwatters**

### CODE REVIEW

File: groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+resource "aws_instance" "abbyy_doc_ocr_ec2" {
+  count                  = var.abbyy_doc_ocr_ec2_count
+  ami                    = var.abbyy_doc_ocr_ami
+  key_name               = aws_key_pair.abbyy_doc_ocr_keypair.key_name
+  instance_type          = var.abbyy_doc_ocr_ec2_instance_size
+  monitoring             = var.monitoring
+  get_password_data      = var.get_password_data
+  subnet_id              = sort(data.aws_subnet_ids.application.ids)[count.index]
+  vpc_security_group_ids = [module.abbyy_doc_ocr_ec2_security_group.this_security_group_id, data.aws_security_group.nagios_shared.id, data.aws_security_group.rdp_shared.id]
```
Comment (MartyFox) [inline groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf:122]: You can remove Nagios from this and future work as its being replaced.
Dont forget to remove the data lookup.

File: groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    {
+      rule                     = "http-80-tcp"
+      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
+    },
+    {
+      from_port                = 10020
+      to_port                  = 10020
+      protocol                 = "tcp"
+      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
+    },
```
Comment (MartyFox) [inline groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf:79]: You can probably replace these with a "self" rule that the module provides, should cover all possible ports between servers in the same security group.

File: groups/live-windows-virtual-machines/data.tf [hunk 3]
```diff
@@ -1,8 +1,86 @@
...
     Name = "vpc-${var.aws_account}"
   }
+}
+
+data "aws_security_group" "nagios_shared" {
+  filter {
+    name   = "group-name"
+    values = ["sgr-nagios-inbound-shared-*"]
+  }
+}
```
Comment (MartyFox) [inline groups/live-windows-virtual-machines/data.tf:18]: Can be removed.

File: groups/live-windows-virtual-machines/keys.tf [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+# ------------------------------------------------------------------------------
+resource "aws_key_pair" "ewok_lic_bac_keypair" {
+  key_name   = "Ewok-License-Backup-Server"
+  public_key = local.ewok_lic_bac_ec2_data["public-key"]
+}
+
+resource "aws_key_pair" "ewok_lic_sec_keypair" {
+  key_name   = "Ewok-License-Second-Server"
+  public_key = local.ewok_lic_sec_ec2_data["public-key"]
+}
```
Comment (MartyFox) [inline groups/live-windows-virtual-machines/keys.tf:12]: These are in the same application stack so can use the same key pair

File: groups/live-windows-virtual-machines/data.tf (hunk 3 above)
Comment (willwatters) [reply groups/live-windows-virtual-machines/data.tf:18]: References to Nagios has been removed.

File: groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf (hunk 1 above)
Comment (willwatters) [reply groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf]: Nagios references removed.

File: groups/live-windows-virtual-machines/keys.tf (hunk 4 above)
Comment (willwatters) [reply groups/live-windows-virtual-machines/keys.tf]: One key used for both servers.

File: groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf (hunk 2 above)
Comment (willwatters) [reply groups/live-windows-virtual-machines/abbyy_doc_ocr_ec2.tf:79]: updated as follows:

ingress_with_self = [
    {
      rule                     = "http-80-tcp"
      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
    },
    {
      from_port                = 10020
      to_port                  = 10020
      protocol                 = "tcp"
      source_security_group_id = module.abbyy_doc_ocr_ec2_security_group.this_security_group_id
    },
    {
      rule                     = "http-80-tcp"
      source_security_group_id = module.abbyy_doc_ocr_internal_alb_security_group.this_security_group_id
    }
  ]

**Review by MartyFox — APPROVED** [review]

**Pull request merged**